### DISCUSSION TITLE
Add Distinctiveness Centrality

### DISCUSSION THREAD

**Discussion by iandreafc** [networkit/networkit#682] [Ideas]
We recently developed a new metric of Social Network Analysis, named **Distinctiveness Centrality**.
The metric gives importance to nodes with distinctive/non-redundant network ties, and is fully described in this paper: https://doi.org/10.1371/journal.pone.0233276
It also has a very reasonable computational complexity.

We are not expert programmers so, for now, we were just able to develop a Python package that calculates Distinctiveness, built upon Networkx: https://github.com/iandreafc/distinctiveness

However, we would love to have the measure directly integrated in Networkit. It would be great if you could help us!

**Comment by hmeyerhenke** [comment]
Dear Andrea, Thanks for your request. We will look into the matter after the next release (expected next week), when we have more resources again. The usual way would be for you to create a pull request -- but there can be intermediate steps of communication in between.

**Comment by iandreafc** [comment]
Thanks a lot @hmeyerhenke !
I am not an expert programmer, so I would not know how to make a proper pull request and not cause damage :) 
Also I know some Python, but cannot program in C++. Here is the code I wrote in Python: https://github.com/iandreafc/distinctiveness
Hopes this helps and in your help! Thanks a lot again 😃

**Comment by BradKML** [comment]
Cross-referencing an aggregate issue on centrality https://github.com/networkit/networkit/discussions/824
